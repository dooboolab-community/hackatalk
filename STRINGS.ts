import * as Localization from 'expo-localization';

import i18n from 'i18n-js';

const en = {
  LOADING: 'Loading...',
  HELLO: 'Hello',
  LOGIN: 'Login',
  EMAIL: 'Email',
  PASSWORD: 'Password',
  SIGN_UP: 'Sign Up',
  LOGOUT: 'Logout',
  FORGOT_PW: 'Forgot password?',
  FIND_PW: 'Find Password',
  NAME: 'name',
  STATUS_MSG: 'Status message',
  REGISTER: 'Register',
  SEND_LINK: 'Send Link',
  FRIEND: 'Friend',
  MESSAGE: 'Message',
  MY_PROFILE: 'My profile',
  UPDATE: 'Update',
  NO_CONTENT: 'No content',
  ADD_FRIEND: 'Add',
  DELETE_FRIEND: 'Delete',
  SEARCH_USER: 'Search User',
  CHAT: 'Chat',
  SEND: 'Send',
  WRITE_MESSAGE: 'Write message...',
  ERROR: 'Error',
  FRIEND_ADDED: 'Added to your friend.',
  FRIEND_ALREADY_ADDED: 'Already your friend.',
  SETTING: 'Setting',
  SIGN_IN_WITH_GOOGLE: 'Sign in with Google',
  SIGN_IN_WITH_FACEBOOK: 'Sign in with Facebook',
};

const ko = {
  LOADING: '로딩중...',
  HELLO: '안녕하세요',
  LOGIN: '로그인',
  LOGOUT: '로그아웃',
  EMAIL: '이메일',
  PASSWORD: '패스워드',
  SIGN_UP: '회원가입',
  FORGOT_PW: '비밀번호를 잊어버리셨나요?',
  FIND_PW: '비밀번호 찾기',
  NAME: '이름',
  STATUS_MSG: '상태메세지',
  REGISTER: '가입',
  SEND_LINK: '링크 보내기',
  FRIEND: '친구',
  MESSAGE: '메세지',
  MY_PROFILE: '나의 프로필',
  UPDATE: '수정',
  NO_CONTENT: '컨텐츠가 없습니다',
  ADD_FRIEND: '친구추가',
  DELETE_FRIEND: '친구삭제',
  SEARCH_USER: '사용자 검색',
  CHAT: '채팅',
  SEND: '보내기',
  WRITE_MESSAGE: '메세지를 입력해주세요…',
  ERROR: '에러',
  FRIEND_ADDED: '친구로 등록되었습니다.',
  FRIEND_ALREADY_ADDED: '이미 친구입니다.',
  SETTING: '설정',
  SIGN_IN_WITH_GOOGLE: '구글 계정으로 시작하기',
  SIGN_IN_WITH_FACEBOOK: '페이스북 계정으로 시작하기',
};

i18n.fallbacks = true;
i18n.translations = { en, ko };
i18n.locale = Localization.locale;

export const getString = (param: string, mapObj?: object): string => {
  if (mapObj) {
    i18n.t(param, mapObj);
  }
  return i18n.t(param);
};
