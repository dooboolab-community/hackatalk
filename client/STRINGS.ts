import * as Localization from 'expo-localization';

import en from './assets/langs/en.json';
import i18n from 'i18n-js';
import ko from './assets/langs/ko.json';

i18n.fallbacks = true;
i18n.translations = { en, ko };
i18n.locale = Localization.locale;

export const getString = (param: string, mapObj?: Record<string, unknown>): string => {
  if (mapObj) {
    return i18n.t(param, mapObj);
  }

  return i18n.t(param);
};
