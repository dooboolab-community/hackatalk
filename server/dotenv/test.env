JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
