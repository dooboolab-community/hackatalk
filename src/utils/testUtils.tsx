import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
