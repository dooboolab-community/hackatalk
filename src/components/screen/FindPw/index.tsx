import * as yup from 'yup';

import { Container, ErrorText, InnerContainer, StyledButtonWrapper, StyledScrollView } from './styles';
import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';
import React, { ReactElement, memo, useCallback } from 'react';

import { Alert } from 'react-native';
import Button from '../../shared/Button';
import TextInput from '../../shared/TextInput';
import { getString } from '../../../../STRINGS';
import useForm from 'react-hook-form';

interface Props {
  navigation?: NavigationScreenProp<NavigationState, NavigationParams>;
}

const schema = yup.object().shape({
  email: yup
    .string()
    .email(getString('EMAIL_FORMAT_NOT_VALID'))
    .required(getString('EMAIL_REQUIRED')),
});

function FindPw(): ReactElement {
  const {
    register,
    errors,
    setValue,
    handleSubmit,
    formState: { touched },
    watch,
    reset,
  } = useForm({
    validationSchema: schema,
  });
  const onSubmit = useCallback(({ email }) => {
    Alert.alert(
      'Find Password',
      `password reset link has been sent to your email: ${email}.`,
    );
    reset({ email: '' });
  }, []);
  const onTextChanged = useCallback((text: string): void | Promise<boolean> => {
    setValue('email', text, true);
  }, []);
  const errorMessage = errors.email && errors.email.message;

  return (
    <Container>
      <StyledScrollView>
        <InnerContainer>
          <TextInput
            testID="findPw_email_input"
            style={{ marginTop: 8 }}
            txtLabel={getString('EMAIL')}
            txtHint={getString('EMAIL')}
            txt={watch('email', '')}
            ref={register({ name: 'email' })}
            onTextChanged={onTextChanged}
            error={errorMessage}
          />
          {!!errors.email && (
            <ErrorText isError={!!errors.email}>
              {errors.email.message}
            </ErrorText>
          )}
          <StyledButtonWrapper>
            <Button
              testID="btnFindPwConfirm"
              isDisabled={!(touched && touched.length) || !!errorMessage}
              onPress={handleSubmit(onSubmit)}
              width="100%"
              containerStyle={{ flexDirection: 'row' }}
            >
              {getString('PASSWORD_RESET')}
            </Button>
          </StyledButtonWrapper>
        </InnerContainer>
      </StyledScrollView>
    </Container>
  );
}

export default memo(FindPw);
