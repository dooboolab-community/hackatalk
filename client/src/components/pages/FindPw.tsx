import {Button, EditText, useTheme} from 'dooboo-ui';
import React, {FC, useState} from 'react';
import type {
  UserFindPwMutation,
  UserFindPwMutationResponse,
} from '../../__generated__/UserFindPwMutation.graphql';
import {showAlertForError, validateEmail} from '../../utils/common';

import {Alert} from 'react-native';
import {AuthStackNavigationProps} from '../navigations/AuthStackNavigator';
import {PayloadError} from 'relay-runtime';
import {findPasswordMutation} from '../../relay/queries/User';
import {getString} from '../../../STRINGS';
import styled from '@emotion/native';
import {useMutation} from 'react-relay/hooks';
import {useNavigation} from '@react-navigation/core';

const Container = styled.View`
  flex: 1;
  padding: 0 32px;
  background-color: ${({theme}) => theme.background};
`;

const ButtonWrapper = styled.View`
  width: 100%;
  margin-top: 20px;
`;

const Page: FC = () => {
  const navigation = useNavigation<AuthStackNavigationProps<'FindPw'>>();
  const [email, setEmail] = useState<string>('');
  const [errorEmail, setErrorEmail] = useState<string>('');

  const [commitFindPassword, isInFlight] = useMutation<UserFindPwMutation>(
    findPasswordMutation,
  );

  const {theme} = useTheme();

  const onFindPw = async (): Promise<void> => {
    if (!validateEmail(email)) {
      setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));

      return;
    }

    const mutationConfig = {
      variables: {email},
      onError: (error: Error) => {
        showAlertForError(error);
      },
      onCompleted: (
        response: UserFindPwMutationResponse,
        errors: PayloadError[] | null,
      ) => {
        const result = response.findPassword;

        if (errors && errors.length > 0) {
          Alert.alert(getString('FAILED'), getString('EMAIL_USER_NOT_EXISTS'));

          return;
        }

        if (result) {
          Alert.alert(
            getString('SUCCESS'),
            getString('PASSWORD_RESET_EMAIL_SENT'),
          );

          navigation.goBack();
        }
      },
    };

    commitFindPassword(mutationConfig);
  };

  return (
    <Container>
      <EditText
        testID="input-email"
        style={{marginVertical: 20}}
        styles={{
          container: {
            borderColor: theme.text,
          },
          input: {
            color: 'red',
          },
        }}
        focusColor={theme.focused}
        labelText={getString('EMAIL')}
        placeholderTextColor={theme.placeholder}
        placeholder="hello@example.com"
        value={email}
        onChangeText={(text: string): void => {
          setEmail(text);
          setErrorEmail('');
        }}
        errorText={errorEmail}
        onSubmitEditing={onFindPw}
      />
      <ButtonWrapper>
        <Button
          testID="btn-find-pw"
          loading={isInFlight}
          onPress={onFindPw}
          styles={{
            container: {
              height: 52,
              backgroundColor: theme.btnPrimary,
              borderWidth: 1,
            },
            text: {
              color: theme.btnPrimaryFont,
            },
            hovered: {
              borderColor: theme.text,
            },
          }}
          text={getString('FIND_PW')}
        />
      </ButtonWrapper>
    </Container>
  );
};

export default Page;
