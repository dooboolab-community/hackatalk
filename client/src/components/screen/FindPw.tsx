import { Button, EditText } from 'dooboo-ui';
import React, { ReactElement, useState } from 'react';
import { showAlertForError, validateEmail } from '../../utils/common';

import { Alert } from 'react-native';
import { AuthStackNavigationProps } from '../navigation/AuthStackNavigator';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/theme';

const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
`;

const Wrapper = styled.View`
  margin: 44px;
`;

const ButtonWrapper = styled.View`
  width: 100%;
  margin-top: 20px;
`;

interface Props {
  navigation: AuthStackNavigationProps<'FindPw'>;
}

function Page({ navigation }: Props): ReactElement {
  const [email, setEmail] = useState<string>('');
  const [errorEmail, setErrorEmail] = useState<string>('');
  const [findingPw, setFindingPw] = useState<boolean>(false);

  const { theme } = useThemeContext();
  // const [findPassword] = useMutation<{ findPassword: boolean }, MutationFindPasswordInput>(MUTATION_FIND_PASSWORD);

  const navigateToSignIn = (): void => navigation.navigate('SignIn');

  const onFindPw = async (): Promise<void> => {
    if (!validateEmail(email)) {
      setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));
      return;
    }

    try {
      setFindingPw(true);
      // const result = await findPassword({ variables: { email } });
      // if (result.data?.findPassword) {
      //   Alert.alert('', getString('PASSWORD_RESET_EMAIL_SENT'), [
      //     {
      //       text: getString('OK'),
      //       onPress: navigateToSignIn,
      //     },
      //   ]);
      // }
    } catch ({ graphQLErrors }) {
      showAlertForError(graphQLErrors);
    } finally {
      setFindingPw(false);
    }
  };

  return (
    <Container>
      <Wrapper>
        <EditText
          testID="input-email"
          errorTestID="error-email"
          textStyle={{
            color: theme.fontColor,
          }}
          borderColor={theme.font}
          focusColor={theme.focused}
          placeholderTextColor={theme.placeholder}
          label={getString('EMAIL')}
          placeholder="hello@example.com"
          value={email}
          onChangeText={(text: string): void => {
            setEmail(text);
            setErrorEmail('');
          }}
          style={{ marginTop: 20 }}
          errorText={errorEmail}
          onSubmitEditing={onFindPw}
        />
        <ButtonWrapper>
          <Button
            testID="btn-find-pw"
            loading={findingPw}
            onPress={onFindPw}
            containerStyle={{ height: 52, backgroundColor: theme.btnPrimary }}
            textStyle={{ color: theme.btnPrimaryFont }}
            text={getString('FIND_PW')}
          />
        </ButtonWrapper>
      </Wrapper>
    </Container>
  );
}

export default Page;
