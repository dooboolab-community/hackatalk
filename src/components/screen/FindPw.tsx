import React, { ReactElement, useState } from 'react';

import { AuthStackNavigationProps } from '../navigation/AuthStackNavigator';
import Button from '../shared/Button';
import { EditText } from '@dooboo-ui/native';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/native-theme';
import { validateEmail } from '../../utils/common';

const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
`;

const Wrapper = styled.View`
  margin: 44px;
`;

const ButtonWrapper = styled.View`
  width: 100%;
  margin-top: 20px;
`;

interface Props {
  navigation: AuthStackNavigationProps<'FindPw'>;
}

function Page(props: Props): ReactElement {
  const [email, setEmail] = useState<string>('');
  const [errorEmail, setErrorEmail] = useState<string>('');
  const [findingPw, setFindingPw] = useState<boolean>(false);
  let timer: number;

  const { theme } = useThemeContext();

  const onFindPw = (): void => {
    if (!validateEmail(email)) {
      setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));
      return;
    }

    setFindingPw(true);
    timer = setTimeout(() => {
      setFindingPw(false);
      clearTimeout(timer);
      if (props.navigation) {
        props.navigation.navigate('SignIn');
      }
    }, 1000);
  };

  return (
    <Container>
      <Wrapper>
        <EditText
          testID="input-email"
          errorTestID="error-email"
          textStyle={{
            color: theme.fontColor,
          }}
          label={getString('EMAIL')}
          placeholder="hello@example.com"
          placeholderTextColor={theme.placeholder}
          value={email}
          onChangeText={(text: string): void => {
            setEmail(text);
            setErrorEmail('');
          }}
          style={{ marginTop: 20 }}
          errorText={errorEmail}
          onSubmitEditing={onFindPw}
        />
        <ButtonWrapper>
          <Button
            testID="btn-find-pw"
            isLoading={findingPw}
            onPress={onFindPw}
            containerStyle={{ padding: 5, height: 52 }}
          >
            {getString('FIND_PW')}
          </Button>
        </ButtonWrapper>
      </Wrapper>
    </Container>
  );
}

export default Page;
