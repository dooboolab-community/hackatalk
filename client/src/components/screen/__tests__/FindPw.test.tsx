import 'react-native';

import React, { ReactElement } from 'react';
import {
  RenderResult,
  act,
  fireEvent,
  render,
  wait,
  waitForElement,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import FindPw from '../FindPw';
import { getString } from '../../../../STRINGS';

// eslint-disable-next-line
let props: any;
let component: ReactElement;

// const mockFindPwMutation = [
//   {
//     request: {
//       query: MUTATION_FIND_PASSWORD,
//       variables: {
//         email: 'email@email.com',
//       },
//     },
//     result: {
//       data: {
//         findPassword: true,
//       },
//     },
//   },
// ];

const mockAlert = {
  alert: jest.fn(),
};

jest.mock('react-native', () => {
  const ReactNative = jest.requireActual('react-native');
  Object.defineProperty(ReactNative, 'Alert', {
    get: () => mockAlert,
  });
  return ReactNative;
});

describe('[FindPw] rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<FindPw {...props} />);
  });

  it('should renders as expected', () => {
    const { baseElement } = render(component);
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('[FindPw] interaction', () => {
  let testingLib: RenderResult;

  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<FindPw {...props} />);
    testingLib = render(component);
  });

  it('should invoke changeText event handler when email changed', () => {
    const textInput = testingLib.getByTestId('input-email');

    act(() => {
      fireEvent.changeText(textInput, 'email@email.com');
    });

    expect(textInput.props.value).toEqual('email@email.com');
  });

  describe('onFindPw', () => {
    beforeAll(() => {
      props = createTestProps();
      component = createTestElement(
        <FindPw {...props} />,
      );
      testingLib = render(component);
    });

    it('should show error text when the email is not validated', () => {
      const textInput = testingLib.getByTestId('input-email');

      act(() => {
        fireEvent.changeText(textInput, 'example');
      });

      const btnFindPw = testingLib.getByTestId('btn-find-pw');
      act(() => {
        fireEvent.press(btnFindPw);
      });
      // const errorText = testingLib.getByTestId('error-email');
      // expect(errorText).toBeTruthy();
    });

    it('should call FindPw when button has clicked and navigate to SignIn', () => {
      const textInput = testingLib.getByTestId('input-email');

      act(() => {
        fireEvent.changeText(textInput, 'email@email.com');
      });

      const btnFindPw = testingLib.getByTestId('btn-find-pw');

      act(() => {
        fireEvent.press(btnFindPw);
      });

      // await wait(() => expect(mockAlert.alert).toHaveBeenCalled());
      // expect(mockAlert.alert.mock.calls[0][1]).toEqual(getString('PASSWORD_RESET_EMAIL_SENT'));
    });

    it('should do nothing when navigation is not defined.', () => {
      props = createTestProps({
        navigation: null,
      });
      component = createTestElement(<FindPw {...props} />);
      testingLib = render(component);

      const textInput = testingLib.getByTestId('input-email');

      act(() => {
        fireEvent.changeText(textInput, 'email@email.com');
      });

      const btnFindPw = testingLib.getByTestId('btn-find-pw');

      act(() => {
        fireEvent.press(btnFindPw);
      });

      expect(props.navigation).toBeNull();
    });
  });
});
