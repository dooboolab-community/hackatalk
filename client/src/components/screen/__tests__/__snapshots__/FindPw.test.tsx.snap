// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FindPw] rendering test should renders as expected 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 44,
              "marginLeft": 44,
              "marginRight": 44,
              "marginTop": 44,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "flexDirection": "column",
              },
              Object {
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#b9b9c4",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "marginBottom": 5,
                },
                undefined,
                false,
              ]
            }
          >
            Email
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            multiline={false}
            placeholder="hello@example.com"
            placeholderTextColor="#676c7a"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "fontSize": 15,
                  "fontWeight": "500",
                  "minHeight": 44,
                  "paddingBottom": 15,
                  "paddingTop": 15,
                },
                Object {
                  "color": "#151A25",
                },
              ]
            }
            testID="input-email"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "width": "100%",
                },
                undefined,
                Object {
                  "borderBottomWidth": 0.6,
                  "borderColor": "#2E2E2E",
                },
                false,
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#1E6EFA",
                  "height": 52,
                },
              ]
            }
          >
            <View
              activeOpacity={0.7}
              delayPressIn={30}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "borderColor": "blue",
                    "borderRadius": 6,
                    "flexDirection": "row",
                    "height": 52,
                    "justifyContent": "center",
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "width": 320,
                  },
                  undefined,
                  false,
                ]
              }
              testID="btn-find-pw"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#069ccd",
                      "fontSize": 14,
                    },
                    Object {
                      "color": "#ffffff",
                    },
                  ]
                }
              >
                Find Password
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?﻿%s﻿%s",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of \`Context.Consumer\`.
    in TouchableOpacity (created by Context.Consumer)
    in StyledNativeComponent (created by Button__StyledButton)
    in Button__StyledButton (created by Button)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by Button__Container)
    in Button__Container (created by Button)
    in Button (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__ButtonWrapper)
    in FindPw__ButtonWrapper (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__Wrapper)
    in FindPw__Wrapper (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__Container)
    in FindPw__Container (created by Page)
    in Page
    in ProfileModalProvider (created by RelayProviderWrapper)
    in Suspense (created by RelayProviderWrapper)
    in RelayEnvironmentProvider (created by RelayProviderWrapper)
    in RelayProviderWrapper (created by AllProviders)
    in AuthProvider (created by AllProviders)
    in ThemeProvider (created by ThemeProvider)
    in ThemeProvider (created by AllProviders)
    in DeviceProvider (created by AllProviders)
    in AllProviders
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)",
                  "substitutions": Array [
                    Object {
                      "length": 48,
                      "offset": 120,
                    },
                    Object {
                      "length": 1533,
                      "offset": 168,
                    },
                  ],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/Data/YellowBoxWarning.js",
                    "lineNumber": 58,
                    "methodName": "Function.parse",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 56,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 180,
                    "methodName": "registerError",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 21,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 84,
                    "methodName": "errorImpl",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 13,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 63,
                    "methodName": "CustomConsole.console.error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 120,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 4,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 92,
                    "methodName": "error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 8,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 8004,
                    "methodName": "validateFunctionComponentInDev",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 6,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 7971,
                    "methodName": "mountIndeterminateComponent",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 9019,
                    "methodName": "beginWork",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 12649,
                    "methodName": "performUnitOfWork",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  

Check the render method of \`Context.Consumer\`.
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  
    in TouchableOpacity (created by Context.Consumer)
    in StyledNativeComponent (created by Button__StyledButton)
    in Button__StyledButton (created by Button)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by Button__Container)
    in Button__Container (created by Button)
    in Button (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__ButtonWrapper)
    in FindPw__ButtonWrapper (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__Wrapper)
    in FindPw__Wrapper (created by Page)
    in View (created by View)
    in View (created by Context.Consumer)
    in StyledNativeComponent (created by FindPw__Container)
    in FindPw__Container (created by Page)
    in Page
    in ProfileModalProvider (created by RelayProviderWrapper)
    in Suspense (created by RelayProviderWrapper)
    in RelayEnvironmentProvider (created by RelayProviderWrapper)
    in RelayProviderWrapper (created by AllProviders)
    in AuthProvider (created by AllProviders)
    in ThemeProvider (created by ThemeProvider)
    in ThemeProvider (created by AllProviders)
    in DeviceProvider (created by AllProviders)
    in AllProviders
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
